internal/fs/vss_windows.go

```diff
@@ -0,0 +1,979 @@
+			uintptr(unsafe.Pointer(vss)), uintptr(unsafe.Pointer(&snapshotID)),
+			uintptr(unsafe.Pointer(properties)))
+	}
+
+	return NewVssErrorIfResultNotOK("GetSnapshotProperties() failed", HRESULT(result))
+}
+
+// vssFreeSnapshotProperties calls the equivalent VSS api.
+func vssFreeSnapshotProperties(properties *VssSnapshotProperties) error {
+	proc, err := findVssProc("VssFreeSnapshotProperties")
+	if err != nil {
+		return err
+	}
+
+	proc.Call(uintptr(unsafe.Pointer(properties)))
+	return nil
+}
+
+// BackupComplete calls the equivalent VSS api.
+func (vss *IVssBackupComponents) BackupComplete() (*IVSSAsync, error) {
+	var oleIUnknown *ole.IUnknown
+	result, _, _ := syscall.Syscall(vss.getVTable().backupComplete, 2, uintptr(unsafe.Pointer(vss)),
+		uintptr(unsafe.Pointer(&oleIUnknown)), 0)
+
+	err := NewVssErrorIfResultNotOK("BackupComplete() failed", HRESULT(result))
+	return vss.convertToVSSAsync(oleIUnknown, err)
+}
+
+// VssSnapshotProperties defines the properties of a VSS snapshot as part of the VSS api.
+type VssSnapshotProperties struct {
+	snapshotID           ole.GUID
+	snapshotSetID        ole.GUID
+	snapshotsCount       uint32
+	snapshotDeviceObject *uint16
+	originalVolumeName   *uint16
+	originatingMachine   *uint16
+	serviceMachine       *uint16
+	exposedName          *uint16
+	exposedPath          *uint16
+	providerId           ole.GUID
+	snapshotAttributes   uint32
+	creationTimestamp    uint64
+	status               uint
+}
+
+// GetSnapshotDeviceObject returns root path to access the snapshot files
+// and folders.
+func (p *VssSnapshotProperties) GetSnapshotDeviceObject() string {
+	return ole.UTF16PtrToString(p.snapshotDeviceObject)
+}
+
+// UIID_IVSS_ASYNC defines to GUID of IVSSAsync.
+var UIID_IVSS_ASYNC = &ole.GUID{
+	0x507C37B4, 0xCF5B, 0x4e95,
+	[8]byte{0xb0, 0xaf, 0x14, 0xeb, 0x97, 0x67, 0x46, 0x7e},
+}
+
+// IVSSAsync VSS api interface.
+type IVSSAsync struct {
+	ole.IUnknown
+}
+
+// IVSSAsync is the vtable for IVSSAsync.
+type IVSSAsyncVTable struct {
+	ole.IUnknownVtbl
+	cancel      uintptr
+	wait        uintptr
+	queryStatus uintptr
+}
+
+// Constants for IVSSAsync api.
+const (
+	VSS_S_ASYNC_PENDING   = 0x00042309
+	VSS_S_ASYNC_FINISHED  = 0x0004230A
+	VSS_S_ASYNC_CANCELLED = 0x0004230B
+)
+
+// getVTable returns the vtable for IVSSAsync.
+func (vssAsync *IVSSAsync) getVTable() *IVSSAsyncVTable {
+	return (*IVSSAsyncVTable)(unsafe.Pointer(vssAsync.RawVTable))
+}
+
+// Wait calls the equivalent VSS api.
+func (vssAsync *IVSSAsync) Cancel() HRESULT {
+	result, _, _ := syscall.Syscall(vssAsync.getVTable().cancel, 1,
+		uintptr(unsafe.Pointer(vssAsync)), 0, 0)
+	return HRESULT(result)
+}
+
+// Wait calls the equivalent VSS api.
+func (vssAsync *IVSSAsync) Wait(millis uint32) HRESULT {
+	result, _, _ := syscall.Syscall(vssAsync.getVTable().wait, 2, uintptr(unsafe.Pointer(vssAsync)),
+		uintptr(millis), 0)
+	return HRESULT(result)
+}
+
+// Wait calls the equivalent VSS api.
```

Comment:
s/Wait/QueryStatus/

Reply:
Fixed in https://github.com/restic/restic/pull/2274/commits/93a55ee3c94ea1f15d3cfe45ae09108764253447